### PR TITLE
Lease Count Quotas: include error message in docs

### PR DESCRIPTION
**Pull request by mpalmi**
This PR adds a breadcrumb for any users seeing the `quota exceeded` error message to guide them on their journey.

### CODE REVIEW

**Review by ccapurso — APPROVED** [review]
Great addition!

File: website/content/docs/enterprise/lease-count-quotas.mdx [hunk 1]
```diff
@@ -86,6 +86,21 @@ cluster through various
 [metrics](/vault/docs/internals/telemetry/metrics/core-system#quota-metrics)
 and through enabling optional audit logging.
 
+## Lease count quota exceeded
+
+When the Lease Count Quota is exceeded, Vault will emit a `429 - Too Many
+Requests`, indicating that new leases are in violation of the quota.
+
+```
+* lease count quota exceeded
```
Comment (mladlow) [inline website/content/docs/enterprise/lease-count-quotas.mdx:95]: I don't know and am not an expert on this but you might need more of the error for it to show up in search results.

Comment (ccapurso) [reply website/content/docs/enterprise/lease-count-quotas.mdx:95]: This is an interesting point. I just tried searching on the Vercel preview and didn't get a hit. I'm not sure if the Vercel previews use the same search index as the deployed doc site though.

File: website/content/docs/enterprise/lease-count-quotas.mdx [hunk 2]
```diff
@@ -86,6 +86,21 @@ cluster through various
 [metrics](/vault/docs/internals/telemetry/metrics/core-system#quota-metrics)
 and through enabling optional audit logging.
 
+## Lease count quota exceeded
+
+When the Lease Count Quota is exceeded, Vault will emit a `429 - Too Many
+Requests`, indicating that new leases are in violation of the quota.
```
Comment (schavis) [inline website/content/docs/enterprise/lease-count-quotas.mdx:92]:
```suggestion
Vault returns a `429 - Too Many Requests` response if a new lease request violates the quota limit.
```

Style correction: write in active voice, simplify sentences

File: website/content/docs/enterprise/lease-count-quotas.mdx [hunk 3]
```diff
@@ -86,6 +86,21 @@ cluster through various
...
+
+```
+* lease count quota exceeded
+```
+
+This error is a call to action to identify the root cause and either:
+
+1. Address any client-side errors leading to excessive lease creation.
+2. Properly tune the Lease Count Quota to accomodate any expected increase in
+   lease creation.
```
Comment (schavis) [inline website/content/docs/enterprise/lease-count-quotas.mdx:102]:
```suggestion
To resolve the error:

1. Check for client-side errors that result in excessive lease creation requests
   and correct them.
1. Tune your lease count quota to accommodate any expected increases in lease
   creation. For example, due to new feature releases or an increase in users.
```

Style correction: avoid "this" as a pronoun, write in active voice, avoid language that suggests users have done something "wrong" (e.g., "properly tune")

**Review by schavis — CHANGES_REQUESTED** [review]

File: website/content/docs/enterprise/lease-count-quotas.mdx (hunk 1 above)
Comment (mpalmi) [reply website/content/docs/enterprise/lease-count-quotas.mdx:94]: Yeah, definitely a good call-out. I'll look into this, thanks @mladlow  (also thank you for the idea to include this in the docs in the first place). 😄

**Comment by mpalmi** [comment]
@mladlow @schavis I wasn't able to validate that this new structure leads to a more searchable error message in the preview, but it seems like it would. Please let me know if you have any recommendations about other things to try.

I also sprinkled in some references to the lease explosions doc.

File: website/content/docs/enterprise/lease-count-quotas.mdx (hunk 1 above)
Comment (mladlow) [reply website/content/docs/enterprise/lease-count-quotas.mdx]: I was thinking more for Google indexing than our indexing, but sadly don't have a good grasp of how it works.

Comment (mpalmi) [reply website/content/docs/enterprise/lease-count-quotas.mdx]: @schavis indicated that the best SEO trick would be to have the exact error message in a top-level heading. I assume having internal links will help with that as well! 🤞

File: website/content/docs/concepts/lease-count-quota-exceeded.mdx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+page_title: Lease count quota exceeded
+description: |-
+  Vault Enterprise error when quota limit is reached.
+---
+
+# Lease count quota exceeded
+
+Vault returns a `429 - Too Many Requests` response if a new lease request
+violates the quota limit, to guard against [lease
+explosions](/vault/docs/concepts/lease-explosions).
```
Comment (schavis) [inline website/content/docs/concepts/lease-count-quota-exceeded.mdx:12]:
```suggestion
explosions](/vault/docs/concepts/lease-explosions):
```

File: website/content/docs/concepts/lease-count-quota-exceeded.mdx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+The error will appear as follows:
```
Comment (schavis) [inline website/content/docs/concepts/lease-count-quota-exceeded.mdx:14]:
```suggestion
```

Comment (schavis) [reply website/content/docs/concepts/lease-count-quota-exceeded.mdx:14]: We don't really need a separate line to introduce the example, you can just put a colon at the end of the previous sentence and use that as the introduction.

**Review by schavis — APPROVED** [review]
Approving so you won't be blocked on a re-review from me, but please merge the suggestion for streamlining the introduction before merging.

**Pull request merged**